project/src/components/card/card.tsx

```diff
@@ -8,15 +8,15 @@ import { CardType } from '../../const';
 type CardProps = {
   offer: Offer;
   cardType: CardType;
-  onCardHover: (id: number | null) => void;
+  onCardHover?: ((id: number | null ) => void) | undefined;
```

Comment:
Знак вопроса уже обозначает что может быть undefined, поэтому не обязательно это писать явно
```suggestion
  onCardHover?: ((id: number | null ) => void);
```